src/Diff/Internal/FingerprintPatcher.php

```diff
@@ -61,24 +61,34 @@ private function patchTermList( TermList $terms, Diff $patch ) {
 	private function patchTerm( TermList $terms, $lang, AtomicDiffOp $diffOp ) {
 		$hasLang = $terms->hasTermForLanguage( $lang );
 
-		if ( $diffOp instanceof DiffOpAdd ) {
-			if ( !$hasLang ) {
-				$terms->setTextForLanguage( $lang, $diffOp->getNewValue() );
-			}
-		} elseif ( $diffOp instanceof DiffOpChange ) {
-			if ( $hasLang
-				&& $terms->getByLanguage( $lang )->getText() === $diffOp->getOldValue()
-			) {
-				$terms->setTextForLanguage( $lang, $diffOp->getNewValue() );
-			}
-		} elseif ( $diffOp instanceof DiffOpRemove ) {
-			if ( $hasLang
-				&& $terms->getByLanguage( $lang )->getText() === $diffOp->getOldValue()
-			) {
-				$terms->removeByLanguage( $lang );
-			}
-		} else {
-			throw new PatcherException( 'Invalid terms diff' );
+		switch ( $diffOp->getType() ) {
+			case 'add':
```

Comment:
Uh, thanks for the pointer. I remember seeing this trick somewhere. I will give it a try.